### PR TITLE
add main and changes in controllers

### PR DESCRIPTION
**Pull request by talbenzur**

### CODE REVIEW

File: src/main/java/App/User.java [hunk 1]
```diff
@@ -40,4 +40,14 @@ public String getPassword() {
     public void setPassword(String password) {
         this.password = password;
     }
+
+    @Override
+    public String toString() {
```
Comment (otabeksherman) [inline src/main/java/App/User.java]: No need to print User data. We are writing it into the file

File: src/main/java/App/client/Main.java [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+        UserController.updateUserName(id, token,"Ron");
+        //update email:
+        UserController.updateUserEmail(id, token,"Ron@gmail.com");
+        //update password:
+        UserController.updateUserPassword(id, token,"Ron123456");
+
+        //print user:
+        System.out.println(user.toString());
+        //delete user:
+        UserController.deleteUser(id, token);
```
Comment (otabeksherman) [inline src/main/java/App/client/Main.java]: The id shouldn't be an argument when we send requests to the UserController, only token and other data if needed

File: src/main/java/App/client/Main.java [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+import App.controller.AuthController;
+import App.controller.UserController;
+
+import java.util.OptionalLong;
 
 public class Main {
     public static void main(String[] args) {
 
+        //register new user
+        User user=AuthController.registerNewUser("Avi","avi@gmail.com","Avi123456");
```
Comment (otabeksherman) [inline src/main/java/App/client/Main.java]: AuthController.registerNewUser should return nothing, only notify the client if registration is successful or not

File: src/main/java/App/service/AuthenticationService.java [hunk 4]
```diff
@@ -12,13 +12,13 @@ public class AuthenticationService {
     private Set<Long> tokenSet = new HashSet<>();
     private TokenService tokenService = new TokenService();
 
-    public void createUser(String name, String email, String password) {
+    public User createUser(String name, String email, String password) {
```
Comment (otabeksherman) [inline src/main/java/App/service/AuthenticationService.java]: Should return void

File: src/main/java/App/controller/UtilController.java [hunk 5]
```diff
@@ -19,14 +19,16 @@ static  boolean checkNameValid(String name){
         }
         return  checkRegex("[A-Z]+[a-z]+",name);
     }
-    static boolean checkPasswordValid(String password) {
+    static boolean checkPasswordValid(String password) {//at least 8 character 1 letter and 1 digit
```
Comment (otabeksherman) [inline src/main/java/App/controller/UtilController.java]: Remove the comment please and add an empty line before the class

**Review by otabeksherman — COMMENTED** [review]
Need some changes which described in the comments

File: src/main/java/App/client/Main.java (hunk 2 above)
Comment (talbenzur) [reply src/main/java/App/client/Main.java]: I think it should

File: src/main/java/App/client/Main.java (hunk 3 above)
Comment (talbenzur) [reply src/main/java/App/client/Main.java]: it is supposed to return the id so we can send it later on update

File: src/main/java/App/controller/UtilController.java (hunk 5 above)
Comment (talbenzur) [reply src/main/java/App/controller/UtilController.java]: I moved the comment next to the phrase itself. so we can know what the regular expression means. Line 28

File: src/main/java/App/client/Main.java (hunk 2 above)
Comment (otabeksherman) [reply src/main/java/App/client/Main.java]: From the requirements:
"When logging in the client receives a Token, this token is used to identify the user when performing actions via the UserController."
Therefore, Client has no access to it's ID and can't perform request by using ID

Comment (talbenzur) [reply src/main/java/App/client/Main.java]: We do receive a token when logging in.
But there is a need for a service comparing ID vs. TOKEN.

File: src/main/java/App/service/AuthenticationService.java (hunk 4 above)
Comment (talbenzur) [reply src/main/java/App/service/AuthenticationService.java]: I changed to Long id

File: src/main/java/App/client/Main.java (hunk 2 above)
Comment (otabeksherman) [reply src/main/java/App/client/Main.java]: There is such service TokenService

Comment (talbenzur) [reply src/main/java/App/client/Main.java]: Look at the changes

**Review by otabeksherman — APPROVED** [review]

**Pull request merged**